src/sort/rational.rs

```diff
@@ -32,15 +34,75 @@ impl Sort for RationalSort {
     #[rustfmt::skip]
     fn register_primitives(self: Arc<Self>, eg: &mut EGraph) {
         type R = BigRational;
+        type Opt<T=()> = Option<T>;
+
         // TODO we can't have primitives take borrows just yet, since it
         // requires returning a reference to the locked sort
         add_primitives!(eg, "+" = |a: R, b: R| -> R { a + b });
         add_primitives!(eg, "-" = |a: R, b: R| -> R { a - b });
         add_primitives!(eg, "*" = |a: R, b: R| -> R { a * b });
-        add_primitives!(eg, "/" = |a: R, b: R| -> R { a / b }); // TODO option
+        add_primitives!(eg, "/" = |a: R, b: R| -> Option<R> {
+            if !b.is_zero() {
+                Some(a / b)
+            } else {
+                None
+            }
+        });
         add_primitives!(eg, "min" = |a: R, b: R| -> R { a.min(b) });
         add_primitives!(eg, "max" = |a: R, b: R| -> R { a.max(b) });
+        add_primitives!(eg, "neg" = |a: R| -> R { -a });
+        add_primitives!(eg, "abs" = |a: R| -> R { a.abs() });
+        add_primitives!(eg, "floor" = |a: R| -> R { a.floor() });
+        add_primitives!(eg, "ceil" = |a: R| -> R { a.ceil() });
+        add_primitives!(eg, "round" = |a: R| -> R { a.round() });
         add_primitives!(eg, "rational" = |a: i64, b: i64| -> R { R::new(a.into(), b.into()) });
+
+        add_primitives!(eg, "pow" = |a: R, b: R| -> Option<R> {
+            if a.is_zero() {
+                if b.is_positive() {
+                    Some(R::zero())
+                } else {
+                    None
+                }
+            } else if b.is_zero() {
+                Some(R::one())
+            } else if b.is_integer() {
+                Some(Pow::pow(a, b.to_integer()))
+            } else {
+                None
+            }
+        });
+        add_primitives!(eg, "log" = |a: R| -> Option<R> {
+            if a.is_one() {
+                Some(R::zero())
+            } else {
+                None
+            }
+        });
```

Comment:
I think we should delete this or replace the body (or even the `else`) with `todo!()` just to be clear it's a stub. Same with `cbrt` below.